tests/app/shared/issue-tables/issues-table-paginated-data.spec.ts

```diff
@@ -0,0 +1,95 @@
+import { MatPaginator, MatPaginatorIntl } from "@angular/material";
+import { Issue } from "../../../../src/app/core/models/issue.model";
+import { TEAM_4 } from "../../../constants/data.constants";
+import {
+    ISSUE_WITH_ASSIGNEES,
+    ISSUE_WITH_EMPTY_DESCRIPTION,
+    ISSUE_WITH_EMPTY_DESCRIPTION_HIGH_SEVERITY,
+    ISSUE_WITH_EMPTY_DESCRIPTION_LOW_SEVERITY
+} from "../../../constants/githubissue.constants";
+import { setAndGetPaginatedData } from '../../../../src/app/shared/issue-tables/issue-tables-paginated-data';
+
+describe('isses-table-paginated-data', () => {
+    describe('getPaginatedData()', () => {
+        const changeDetectorRefMock = {
+            detectChanges: () => null,
+            markForCheck: () => null,
+            detach: () => null,
+            reattach: () => null,
+            checkNoChanges: () => null
+        };
+        let dataSet_3: Issue[];
+        let dataSet_7: Issue[];
+        let paginator: MatPaginator;
+        const dummyTeam = TEAM_4;
+        const mediumSeverityIssueWithResponse: Issue = Issue.createPhaseTeamResponseIssue(ISSUE_WITH_EMPTY_DESCRIPTION, dummyTeam);
+        const mediumSeverityIssueWithAssigneee: Issue = Issue.createPhaseTeamResponseIssue(ISSUE_WITH_ASSIGNEES, dummyTeam);
+        const lowSeverityFeatureFlawIssue: Issue = Issue.createPhaseTeamResponseIssue(ISSUE_WITH_EMPTY_DESCRIPTION_LOW_SEVERITY, dummyTeam);
+        const highSeverityDocumentationBugIssue: Issue =
+            Issue.createPhaseTeamResponseIssue(ISSUE_WITH_EMPTY_DESCRIPTION_HIGH_SEVERITY, dummyTeam);
+
+        beforeEach(() => {
+            dataSet_3 = [
+                mediumSeverityIssueWithResponse,
+                mediumSeverityIssueWithAssigneee,
+                lowSeverityFeatureFlawIssue
+            ];
+            dataSet_7 = [
+                mediumSeverityIssueWithResponse,
+                mediumSeverityIssueWithAssigneee,
+                lowSeverityFeatureFlawIssue,
+                mediumSeverityIssueWithResponse,
+                mediumSeverityIssueWithAssigneee,
+                lowSeverityFeatureFlawIssue,
+                highSeverityDocumentationBugIssue
+            ];
+            paginator = new MatPaginator(new MatPaginatorIntl(), changeDetectorRefMock);
```

Comment:
For the purpose of these tests, let's use a literal object for the `paginator` (as it's simpler):
```suggestion
            paginator = { pageSize: 3 } as MatPaginator;
```